### PR TITLE
Initialize target

### PR DESCRIPTION
**Pull request by rabernat**
This PR creates a new transform called `PrepareZarrTarget`. This is the transform which takes a dataset schema (now formalized as an `XarraySchema` TypedDict) and uses it to initialize a Zarr store. This is a crucial step for writing data to the target.

### CODE REVIEW

File: tests/test_options.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                assert item["cache_url"] == "bar"
+
+        return _check_results
+
+    pipeline_options = PipelineOptions(target_url="foo", cache_url="bar")
+    custom_options = pipeline_options.view_as(PangeoForgeOptions)
+
+    with TestPipeline(options=custom_options) as p:
+        pcol = p | beam.Create([None]) | beam.Map(get_options)
+        assert_that(pcol, has_correct_options())
```
Comment (rabernat) [inline tests/test_options.py:29]: This test does not work. The options are not getting set. Can any Beam expert provide some advice on the right way to do this?

Comment (rabernat) [reply tests/test_options.py]: The latest commits show a pattern of passing options that appears to work.

**Review by alxmrs — COMMENTED** [review]
A few notes, mostly nits.

File: pangeo_forge_recipes/aggregation.py [hunk 2]
```diff
@@ -29,6 +32,9 @@ class XarrayCombineAccumulator:
     def add_input(self, s: dict, position: int) -> None:
         s = s.copy()  # avoid modifying input
         if "chunks" not in s:
+            # By default, the output of ds.to_dict(data=False) includes nothing
```
[the remark this answers is not in the document]
Comment (cisaacstern) [reply pangeo_forge_recipes/aggregation.py:35]: This is a cool PEP! Hadn't seen it before, thanks or sharing Alex!

[the remark this answers is not in the document]
Comment (rabernat) [reply pangeo_forge_recipes/aggregation.py:35]: That's a great idea! Or even better, we could use https://github.com/carbonplan/xarray-schema/

[the remark this answers is not in the document]
Comment (rabernat) [reply pangeo_forge_recipes/aggregation.py]: I have gone with TypedDict in 1973464. In the future we can consider adding xarray-schema as a dependency. It seems to have a slightly different aim (validation) than what we need here.

**Pull request merged**